src/components/Header/Header.js

```diff
@@ -0,0 +1,37 @@
+import { MDCTopAppBar } from "@material/top-app-bar";
+import headerTemplate from "./Header.html";
+
+export class HeaderComponent {
+  constructor(mountPoint, props = {}) {
+    this.mountPoint = mountPoint;
+    this.openFunction = props.openFunction;
+    this.scrollTarget = props.scrollTarget;
+  }
+
+  initMaterial() {
+    this.topAppBar = MDCTopAppBar.attachTo(document.querySelector(".header"));
+  }
+
+  querySelectors() {
+    this.topAppBar.setScrollTarget(this.scrollTarget);
+  }
+
+  addEventListeners() {
+    this.topAppBar.listen("MDCTopAppBar:nav", this.onOpen.bind(this));
+  }
+
+  onOpen(e) {
```

Comment:
no need for this method - you can directly use callback from props inside 

`this.topAppBar.listen("MDCTopAppBar:nav"   ...`